lib/article_json/utils/additional_element_placer.rb

```diff
@@ -47,6 +47,7 @@ def merge_elements
           .each_cons(2)
           .each_with_object([]) do |(element, next_element), result|
             result << element
+            next if remaining_elements.empty?
             characters_passed += element.length if element.respond_to?(:length)
             next_in -= element.length if element.respond_to?(:length)
```

Comment:
I noticed that these two lines could also be:
```
  if element.respond_to?(:length)
    characters_passed += element.length
    next_in -= element.length
  end
```
to reduce complexity.

Reply:
Yup, that sounds like a good idea :)